src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const fromParts = fromFormat.slice(0, 3);
+  const toParts = toFormat.slice(0, 3);
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  for (let i = 0; i < fromParts.length; i++) {
+    dateMap[fromParts[i]] = dateParts[i];
+  }
```

Comment:
The for loop used to populate `dateMap` is clear, but you can also consider using the array method `forEach` for a more functional approach, which can improve readability and align with modern JavaScript practices.